### PR TITLE
feat(explore): Time picker enhancement follow up

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY
[follow up PR](https://github.com/apache/incubator-superset/pull/11418)
[UI Design](https://www.figma.com/file/JWaGztdhZS0kS5ruG7x9tB/Control-Panel?node-id=195%3A36740)

- Break down control to child components
- Extract frontend utilities to a separate module and add unit tests(unit test to be added soon)
- Change modal to popover
- Improve typescript typing
- Remove the validate button
- refactor code smells / fix cosmetic issue
- added cypress

## Before
image

#### Last
image

#### Previous
image

#### Custom
image

#### Advanced
image

#### No filter

image

### TEST PLAN

Added cypress for the time picker

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by adam-stasiak** [comment]
Please take a care of flaky tests 
image
You can see more at [link](https://dashboard.cypress.io/projects/ukwxzo/analytics/flaky-tests?branches=%5B%5D&browsers=%5B%5D&chartRangeMostCommonErrors=%5B%5D&chartRangeSlowestTests=%5B%5D&chartRangeTopFailures=%5B%5D&committers=%5B%5D&cypressVersions=%5B%5D&flaky=%5B%5D&operatingSystems=%5B%5D&runGroups=%5B%5D&specFiles=%5B%5D&status=%5B%7B%22label%22%3A%22Passed%22%2C%22value%22%3A%22PASSED%22%7D%2C%7B%22label%22%3A%22Failed%22%2C%22value%22%3A%22FAILED%22%7D%5D&tags=%5B%5D&timeInterval=WEEK&timeRange=%7B%22startDate%22%3A%222020-12-05%22%2C%22endDate%22%3A%222021-01-04%22%7D&viewBy=TEST_CASE)

File: superset-frontend/src/explore/components/controls/DateFilterControl/utils.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+ * Does not do any sort of date validation, only checks if the string is according to the ISO 8601 spec.
+ *  YYYY-MM-DDThh:mm:ss
+ *  YYYY-MM-DDThh:mm:ssTZD
+ *  YYYY-MM-DDThh:mm:ss.sTZD
+ * @see: https://www.w3.org/TR/NOTE-datetime
+ */
+const iso8601 = String.raw`\d{4}-\d\d-\d\dT\d\d:\d\d:\d\d(?:\.\d+)?(?:(?:[+-]\d\d:\d\d)|Z)?`;
+const datetimeConstant = String.raw`TODAY|NOW`;
+const grainValue = String.raw`[+-]?[1-9][0-9]*`;
+const grain = String.raw`YEAR|QUARTER|MONTH|WEEK|DAY|HOUR|MINUTE|SECOND`;
```
Comment (ktmud) [inline superset-frontend/src/explore/components/controls/DateFilterControl/utils.ts:39]: Any reason to use normal case for these constants?

File: superset-frontend/src/explore/components/controls/DateFilterControl/AdvancedFrame.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ */
+import React from 'react';
+import { t } from '@superset-ui/core';
+import { SEPARATOR } from 'src/explore/dateFilterUtils';
+import { Input } from 'src/common/components';
+
+type AdvancedFrameProps = {
+  onChange: (timeRange: string) => void;
+  value: string;
+};
```
Comment (ktmud) [inline superset-frontend/src/explore/components/controls/DateFilterControl/AdvancedFrame.tsx:27]: Nit: for components with very few props, maybe you can just inline the typing and use object deconstrution.

```ts
export default function AdvancedFrame({ onChange, value }: {
  onChange: (timeRange: string) => void;
  value: string;
}) {
```

However, I think all the different subcomponents would have the same signature, so you should probably reuse the `Props` type. E.g. add a `FrameComponentProps` to `types.ts`.

I'd also recommend create another subfolder to make the file directories easier to glance.

**Review by ktmud — APPROVED** [review]
Mostly LGTM. A couple of more suggestions on code structure.

Do you plan to add backend validation back in any shape or form? E.g., add a loading state when clicking on "Apply" and block the popover from closing when validation fails.

File: superset-frontend/src/explore/components/controls/DateFilterControl/AdvancedFrame.tsx (hunk 2 above)
Comment (zhaoyongjie) [reply superset-frontend/src/explore/components/controls/DateFilterControl/AdvancedFrame.tsx:27]: Thank you for the reminder.  I will change this typing to inline.

File: superset-frontend/src/explore/components/controls/DateFilterControl/utils.ts (hunk 1 above)
Comment (zhaoyongjie) [reply superset-frontend/src/explore/components/controls/DateFilterControl/utils.ts:39]: Use UPPER CASE just for constant looks well. In fact, case insensitive in python datetime expression parser, in other words 'YEAR' == 'year' == 'YEAR' for backend datetime parser.

**Comment by zhaoyongjie** [comment]
> Please take a care of flaky tests
> image
> You can see more at [link](https://dashboard.cypress.io/projects/ukwxzo/analytics/flaky-tests?branches=%5B%5D&browsers=%5B%5D&chartRangeMostCommonErrors=%5B%5D&chartRangeSlowestTests=%5B%5D&chartRangeTopFailures=%5B%5D&committers=%5B%5D&cypressVersions=%5B%5D&flaky=%5B%5D&operatingSystems=%5B%5D&runGroups=%5B%5D&specFiles=%5B%5D&status=%5B%7B%22label%22%3A%22Passed%22%2C%22value%22%3A%22PASSED%22%7D%2C%7B%22label%22%3A%22Failed%22%2C%22value%22%3A%22FAILED%22%7D%5D&tags=%5B%5D&timeInterval=WEEK&timeRange=%7B%22startDate%22%3A%222020-12-05%22%2C%22endDate%22%3A%222021-01-04%22%7D&viewBy=TEST_CASE)

I will keep an eye on this flaky test

**Pull request merged**